cpp/src/io/parquet/writer_impl.cu

```diff
@@ -171,7 +171,7 @@ struct aggregate_writer_metadata {
     std::vector<std::vector<uint8_t>> column_indexes;
   };
   std::vector<per_file_metadata> files;
-  std::string created_by                                   = "";
+  std::string created_by = "cuDF Version " CUDF_STRINGIFY(CUDF_VERSION);
```

Comment:
what do other libraries write here?

Reply:
stuff like "parquet-cpp-arrow version 12.0.1" or "parquet-mr version 1.11.1 (build 765bd5cd7fdef2af1cecd0755000694b992bfadd)"

Reply:
Should we align with those libraries on using `version` (lowercase) instead of `Version` (uppercase)? What about `cuDF` vs. `cudf` vs. `libcudf`?

Reply:
I'm good with any of those...just checked arrow-rs and they also use lower case "version", so I'll change that.  I wasn't sure what to use for the name..."libcudf" was my first choice actually. I'll go with that if there are no objections.

Reply:
Apparently the parquet spec actually specifies a format :open_mouth: 

`<Application> version <App Version> (build <App Build Hash>)`

Is there a build hash available anywhere? I found a suggestion to use  `git log -1 --format=%H`. I could add that to the properties for building `writer_impl.cu`.

Reply:
I'm okay with punting on build hashes. How do you feel about that @vuule? If it's important we can inject some CMake define for `CUDF_BUILD_HASH` and use `CUDF_STRINGIFY(CUDF_BUILD_HASH)` or similar.

Reply:
`cudf` is my preference, after further consideration. Python libraries will also call this code, and there is only one "cudf" implementation. Also the other libraries aren't including a "lib" prefix for compiled libraries, so it might be more natural to call it `cudf`.

Reply:
I'm okay with cuDF or cudf. FWIW, the ORC specs call us "CUDF".

Reply:
How about cudF....kinda looks like a race car then :rofl: